client/engine/store/memstore.go

```diff
@@ -51,6 +51,11 @@ func (o *syncMap[T]) Store(key string, data T) {
 	o.m.Store(key, data)
 }
 
+// Delete cleats the value for a key.
+func (o *syncMap[T]) Delete(key string) {
```

Comment:
Is there a case to be made for `Archive` rather than `Delete`?

Reply:
Is `Archive` an actual, or imagined function? And are you suggesting a different behaviour of just a different name?

Of course some kind of "soft delete' would seem safest... but I think unnecessary because replacement by the other type should preserve all of the important information.

Reply:
Yes, I'm imagining `Archive` as a soft-delete. More like adding a `deleted: true` flag.